pkg/storage/s3/s3.go

```diff
@@ -936,7 +945,7 @@ func (is *ObjectStorage) FinishBlobUpload(repo, uuid string, body io.Reader, dig
 	is.Lock(&lockLatency)
 	defer is.Unlock(&lockLatency)
 
-	if is.dedupe && is.cache != nil {
+	if is.dedupe && fmt.Sprintf("%v", is.cache) != fmt.Sprintf("%v", nil) {
```

Comment:
fmt.Sprintf() in the fast path?
also what is this check supposed to do? is.cache != nil check doesn't suffice?

Reply:
> is.cache != nil check doesn't suffice?
We ran into this a while back, it's a bit of a pickle. is.cache is now an interface, and there's a catch there: when you check if an interface == nil, you check its type, not its value, and since it's not a nil interface, the check fails. You need to grab the underlying value, and you either do that with reflect, or with the hackish method you see there. 
More about this https://trstringer.com/go-nil-interface-and-interface-with-nil-concrete-value/